src/WebJobs.Extensions.DurableTask/AzureStorageAccountProvider.cs

```diff
@@ -44,7 +49,11 @@ public StorageAccountDetails GetStorageAccountDetails(string connectionName)
             AzureStorageAccountOptions account = connectionInfo.Get<AzureStorageAccountOptions>();
             if (account != null)
             {
-                TokenCredential credential = this.credentialFactory.Create(connectionInfo);
+                string key = !string.IsNullOrEmpty(account.AccountName) ? account.AccountName : account.BlobServiceUri.ToString();
```

Comment:
I added this since customers can enable managed identity using 2 different set of settings:

```
AzureWebJobsStorage__accountName: For example: mystorageaccount123
```

OR
```
AzureWebJobsStorage__blobServiceUri: Example: https://mystorageaccount123.blob.core.windows.net/

AzureWebJobsStorage__queueServiceUri: Example: https://mystorageaccount123.queue.core.windows.net/

AzureWebJobsStorage__tableServiceUri: Example: https://mystorageaccount123.table.core.windows.net/
```

Reply:
I thought it should be `connectionName` too, but then realized that it was always the default value "Storage". I see the argument to keep it `connectionName` though since `connectionInfo` is the input for `credential`.

Reply:
If `connectionName` is the value from which `accountName` and `blobServiceUri` are derived, then I think it's reasonable to use it as the key. @bachuv - is there a particular concern that you have with it always having "Storage" as the default value?

Reply:
> If connectionName is the value from which accountName and blobServiceUri are derived

It looks like `accountName` and `blobServiceUri` come from `connectionInfo` actually. Let me know if I misunderstood your comment.

>  @bachuv - is there a particular concern that you have with it always having "Storage" as the default value?

I don't have any concerns about always having "Storage" as the default value. I mostly just wanted to point out that I don't see us using `connectionName` in the managed identity code path.

Reply:
How connection info works is that it will retrieve the config section for `AzureWebJobs{ConnectionName}`. In this case, you are getting `AzureWebJobsStorage` section, which includes blobServiceUri and all that. Since this is an `IConfiguration` object, it _does_ include all the managed identity information (`AzureStorageAccountOptions` is just a subset of what is in that section) and that is what the `ITokenCredentialFactory` ends up using.

So, the `connectionName`, ultimately maps to the config section which contains all of the managed identity properties, is the correct key to use.